controller/comments.go

```diff
@@ -137,13 +137,17 @@ func (c *CommentsController) Delete(ctx *app.DeleteCommentsContext) error {
 	}
 	var cm *comment.Comment
 	var wi *workitem.WorkItem
+	var userIsCreator bool
 	// Following transaction verifies if a user is allowed to delete or not
 	err = application.Transactional(c.db, func(appl application.Application) error {
 		cm, err = appl.Comments().Load(ctx.Context, ctx.CommentID)
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-
+		if *identityID == cm.CreatedBy {
+			userIsCreator = true
+			return nil
```

Comment:
Looks correct to me. The work item is needed only if the user is no the comment creator. In that case we need to load the work item to get the space ID to check if the user is a space collaborator.